### PR TITLE
remove param finish

### PR DESCRIPTION
**Pull request by zixuanzh**
In this PR,

- Commented out PARAM_FINISH but did not remove to keep track of parameters and functions that need to be set so that it can be run
- Implemented big int operation for slashing reward
- Proportion related calculation, specifically related to power, is left as zero, they will be fixed in a separate PR when the power table is removed as discussed

### CODE REVIEW

File: actors/builtin/storage_power/policy.go [hunk 1]
```diff
@@ -34,16 +35,18 @@ func pledgePenaltyForConsensusFault(currPledge abi.TokenAmount, faultType Consen
...
 func rewardForConsensusSlashReport(elapsedEpoch abi.ChainEpoch, collateralToSlash abi.TokenAmount) abi.TokenAmount {
-	autil.TODO()
-	// BigInt Operation
-	// var growthRate = builtin.SLASHER_SHARE_GROWTH_RATE_NUM / builtin.SLASHER_SHARE_GROWTH_RATE_DENOM
-	// var multiplier = growthRate^elapsedEpoch
-	// var slasherProportion = min(INITIAL_SLASHER_SHARE * multiplier, 1.0)
-	// return collateralToSlash * slasherProportion
-	return abi.NewTokenAmount(0)
+	// autil.PARAM_FINISH()
+	growthRate := big.Div(indices.ConsensusFault_SlasherShareGrowthRateNum(), indices.ConsensusFault_SlasherShareGrowthRateDenom())
```
Comment (anorth) [inline actors/builtin/storage_power/policy.go]: This evaluates to a constant `1`. The point of separating numerator and denominator is to re-arrange the arithmetic to do division last. Please document the canonical expression in a comment, and then code the arithmetic to work with integers.

File: actors/builtin/storage_power/policy.go [hunk 2]
```diff
@@ -34,16 +35,18 @@ func pledgePenaltyForConsensusFault(currPledge abi.TokenAmount, faultType Consen
...
+	multiplier := big.NewInt(1)
+	for i := int64(0); i < int64(elapsedEpoch); i++ {
+		multiplier = big.Mul(multiplier, growthRate)
+	}
+	initial_slasher_share := big.Div(indices.ConsensusFault_SlasherInitialShareNum(), indices.ConsensusFault_SlasherInitialShareDenom())
```
Comment (anorth) [inline actors/builtin/storage_power/policy.go]: This is a constant `0`.

File: actors/builtin/storage_power/policy.go [hunk 3]
```diff
@@ -34,16 +35,18 @@ func pledgePenaltyForConsensusFault(currPledge abi.TokenAmount, faultType Consen
...
+	slasher_proportion := big.Min(big.Mul(initial_slasher_share, multiplier), big.NewInt(1))
+	return big.Mul(collateralToSlash, slasher_proportion)
 }
 
 func consensusPowerForWeight(weight SectorStorageWeightDesc) abi.StoragePower {
-	autil.PARAM_FINISH()
-	panic("")
+	// autil.PARAM_FINISH()
+	return big.Mul(big.NewInt(int64(weight.SectorSize)), big.NewInt(int64(weight.Duration)))
```
Comment (anorth) [inline actors/builtin/storage_power/policy.go]: This looks wrong. Power is measured in bytes, not byte-epochs.

File: actors/builtin/storage_power/policy.go [hunk 4]
```diff
@@ -1,26 +1,27 @@
...
 	abi "github.com/filecoin-project/specs-actors/actors/abi"
 	builtin "github.com/filecoin-project/specs-actors/actors/builtin"
-	autil "github.com/filecoin-project/specs-actors/actors/util"
+	indices "github.com/filecoin-project/specs-actors/actors/runtime/indices"
 )
 
 // TerminationFault
 func pledgePenaltyForSectorTermination(weight SectorStorageWeightDesc, termType builtin.SectorTermination) abi.TokenAmount {
-	autil.PARAM_FINISH()
+	// autil.PARAM_FINISH()
```
Comment (anorth) [inline actors/builtin/storage_power/policy.go]: Remove the `autil` and `()` from there everywhere. We can just use the token `PARAM_FINISH` in a comment. to mark these places

File: actors/runtime/indices/indices.go [hunk 5]
```diff
@@ -146,25 +133,26 @@ func (inds *IndicesImpl) SectorWeight(
 ) abi.SectorWeight {
 	// for every sector, given its size, start, end, and deals within the sector
 	// assign sector power for the duration of its lifetime
-	PARAM_FINISH()
-	panic("")
+	// PARAM_FINISH()
+	sectorDuration := endEpoch - startEpoch
+	return abi.SectorWeight(big.NewInt(int64(sectorDuration) * int64(sectorSize)))
```
Comment (anorth) [inline actors/runtime/indices/indices.go]: This is the wrong units. Just return sectorSize if you want to ignore deal weight for now.

File: actors/runtime/indices/indices.go [hunk 6]
```diff
@@ -181,22 +169,23 @@ func (inds *IndicesImpl) StoragePower(
...
 }
 
 func (inds *IndicesImpl) CurrEpochBlockReward() abi.TokenAmount {
 	// total block reward allocated for CurrEpoch
 	// each expected winner get an equal share of this reward
 	// computed as a function of NetworkKPI, NetworkTarget, LastEpochReward, TotalUnmminedFIL, etc
-	PARAM_FINISH()
-	panic("")
+	// PARAM_FINISH()
+	return abi.NewTokenAmount(0)
```
Comment (anorth) [inline actors/runtime/indices/indices.go]: At least make this non-zero so a network could progress.

Also ⚠️ here is a function that does not have the right type, and fails to declare the parameters it needs.

File: actors/runtime/indices/indices.go [hunk 7]
```diff
@@ -214,68 +203,67 @@ func (inds *IndicesImpl) NetworkTransactionFee(
...
 }
 
 func (inds *IndicesImpl) GetCurrBlockRewardForMiner(
 	minerStoragePower abi.StoragePower,
 	minerPledgeCollateral abi.TokenAmount,
 ) abi.TokenAmount {
-	PARAM_FINISH()
-	panic("")
+	// PARAM_FINISH()
+	return abi.NewTokenAmount(0)
```
Comment (anorth) [inline actors/runtime/indices/indices.go]: Non-zero please

File: actors/runtime/indices/indices.go (hunk 5 above)
Comment (zixuanzh) [reply actors/runtime/indices/indices.go]: Understood, it is unresolved how SectorWeight should be computed and integral of space over time can be an option. Will use sectorSize for now.

File: actors/builtin/storage_power/policy.go (hunk 3 above)
Comment (zixuanzh) [reply actors/builtin/storage_power/policy.go]: Similar to comment at SectorWeight, but will revert to just SectorSize for now

File: actors/runtime/indices/indices.go (hunk 6 above)
Comment (zixuanzh) [reply actors/runtime/indices/indices.go]: ack, per input type, we were assuming this is computed from the state tree before but I think we can update the input type here now.

Comment (zixuanzh) [reply actors/runtime/indices/indices.go]: will leave out the details for a future PR

File: actors/runtime/indices/indices.go [hunk 8]
```diff
@@ -82,6 +83,8 @@ type IndicesImpl struct {
 	// they are treated as globally available states
 	Epoch                      abi.ChainEpoch
 	NetworkKPI                 big.Int
+	NetworkIndicator           big.Int
+	NetworkTime                big.Int
```
Comment (anorth) [inline actors/runtime/indices/indices.go]: Well those are opaque! Please document.

Comment (zixuanzh) [reply actors/runtime/indices/indices.go]: will document this in a separate PR, thanks for flagging

**Pull request merged**